File: frontend/src/features/admin/dtos/AuditLogDTO.ts
export type PrimitiveValueDTO = string | number | boolean | null;

export type ModifiedValueDTO= {
  Old: PrimitiveValueDTO;
  New: PrimitiveValueDTO;
};

export type AuditLogDTO = {
  entityName: string;
  dateTime: string | Date;
  action: "Added" | "Modified" | "Deleted" | string;
  changes: Record<string, PrimitiveValueDTO | ModifiedValueDTO> | null;
  user: string;
};

File: Backend/Application/DTOs/Audit/AuditLogDTO.cs
namespace Backend.Application.DTOs.Audit
{
    public class AuditLogDTO
    {
        public string EntityName { get; set; } = null!;
        public DateTime DateTime { get; set; }
        public string Action { get; set; } = null!;
        public object Changes { get; set; } = null!;
        public string User { get; set; } = null!;
    }  
}
